.github/workflows/release.yml

```diff
@@ -68,6 +91,9 @@ jobs:
           name: wheels for py3.${{ matrix.python }} on ${{ matrix.os }}
           path: dist/*
       - name: Publish package
+        if: |
+          github.event_name == 'release'
+          && github.repository == 'dask-contrib/dask-sql'
```

Comment:
Minor nit: It might make sense to consolidate this and the case below to a common condition that sets an environment variable that can be picked up here and below